webapp/tbapi/tbapi/blueprints/api.py

```diff
@@ -1,20 +1,137 @@
-from flask import Blueprint, render_template, make_response, jsonify, flash, redirect, url_for, request, current_app
+from flask import Flask, Blueprint, render_template, make_response, jsonify, flash, redirect, url_for, request, current_app
 from tbapi.models.mpower import Patient, User
+from flask_swagger import swagger
+from tbapi.models.tbapp import Note
 
 bp = Blueprint('api', __name__)
 
-@bp.route('/patients')
-def patient_index():
-    return jsonify(patients=[i.serialize for i in Patienst.query.all()])
+@bp.route('/notes/<int:id>', methods=['GET']) # note id
+@bp.route('/notes/', methods=['GET']) # patient_id
+def get(id=None):
+    if (id is not None) & isinstance(id, int): # note id
+      return jsonify(notes=[i.serialize for i in Note.get_by_note(id)])
+  
+    try:
+        patient_id = int(request.args.get('patient_id'))
+    except:
+        return jsonify(notes=[i.serialize for i in Note.get_by_patient_id(0)])
+  
+    if (patient_id is not None) & isinstance(patient_id, int): # patient_id
+        return jsonify(notes=[i.serialize for i in Note.get_by_patient_id(patient_id)])
+    else: # param is none
+        return jsonify(notes=[i.serialize for i in Note.get_by_patient_id(0)])
 
-@bp.route('/patients/<int:id>')
-def patient(id):
-    return jsonify(Patient.query.filter_by(id=id).first().serialize)
+    """Access basics for patient notes
 
-@bp.route('/users')
-def user_index():
-    return jsonify(users=[i.serialize for i in User.query.all()])
+    1. GET /api/notes/?patient_id=int  returns all notes by specified patient_id
+    2. GET /api/notes/{int:id} returns note specified by id.
 
-@bp.route('/users/<int:id>')
-def user(id):
-    return jsonify(User.query.filter_by(id=id).first().serialize)
+    returns patient notes id, patient_id, text, author_id, created, lastmod in JSON
+    ---
+    tags:
+        - Note GET
+    operationId: getNotes
+    parameters:
+      - in: body
+        name: body
+        schema:
+          id: note
+          properties:
+              id:
+                  type: integer
+                  format: int64
+                  description: id for note
+              patient_id:
+                  type: integer
+                  description: patient_id
+              text:
+                  type: string
+                  description: body of note
+              author_id:
+                  type: integer
+                  description: author_id
+              created:
+                  type: string
+                  format: date-time
+                  description: created
+              lastmod:
+                  type: string
+                  format: date-time
+                  description: lastmod
```

Comment:
Here too!

Reply:
discussion at 10/9/2018 stand up regarding passing GMT from the client